### ISSUE TITLE
bug with newer pydantic

### ISSUE THREAD

**Issue by vsoch** [sphinx-doc/sphinx#11479]
### Describe the bug

updating to pydantic 2.0 (or anything higher than 1.10.8 I see:

```console
Running Sphinx v7.0.1

Exception occurred:
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 163, in apply_each_item_validators
    raise TypeError(
TypeError: `@validator(..., each_item=True)` cannot be applied to fields with a schema of tuple-positional
The full traceback has been saved in /tmp/sphinx-err-b3dqvs3z.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:20: html] Error 2
```
And the error log:

```
$ cat /tmp/sphinx-err-b3dqvs3z.log 
# Platform:         linux; (Linux-5.15.0-75-generic-x86_64-with-glibc2.35)
# Sphinx version:   7.0.1
# Python version:   3.9.12 (CPython)
# Docutils version: 0.20.1
# Jinja2 version:   3.1.2
# Pygments version: 2.15.1

# Last messages:


# Loaded extensions:


# Traceback:
Traceback (most recent call last):
  File "/tmp/docs/env/lib/python3.9/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/tmp/docs/env/lib/python3.9/site-packages/sphinx/application.py", line 229, in __init__
    self.setup_extension(extension)
  File "/tmp/docs/env/lib/python3.9/site-packages/sphinx/application.py", line 402, in setup_extension
    self.registry.load_extension(self, extname)
  File "/tmp/docs/env/lib/python3.9/site-packages/sphinx/registry.py", line 442, in load_extension
    metadata = setup(app)
  File "/tmp/docs/env/lib/python3.9/site-packages/sphinx_immaterial/__init__.py", line 313, in setup
    app.setup_extension("sphinx_immaterial.custom_admonitions")
  File "/tmp/docs/env/lib/python3.9/site-packages/sphinx/application.py", line 402, in setup_extension
    self.registry.load_extension(self, extname)
  File "/tmp/docs/env/lib/python3.9/site-packages/sphinx/registry.py", line 429, in load_extension
    mod = import_module(extname)
  File "/home/vanessa/anaconda3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/tmp/docs/env/lib/python3.9/site-packages/sphinx_immaterial/custom_admonitions.py", line 58, in <module>
    class CustomAdmonitionConfig(pydantic.BaseModel):
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_model_construction.py", line 172, in __new__
    complete_model_class(
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_model_construction.py", line 420, in complete_model_class
    schema = cls.__get_pydantic_core_schema__(cls, handler)
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/main.py", line 533, in __get_pydantic_core_schema__
    return __handler(__source)
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_schema_generation_shared.py", line 82, in __call__
    schema = self._handler(__source_type)
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 264, in generate_schema
    return self._generate_schema_for_type(
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 285, in _generate_schema_for_type
    schema = self._generate_schema(obj)
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 475, in _generate_schema
    return self._model_schema(obj)
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 351, in _model_schema
    {k: self._generate_md_field_schema(k, v, decorators) for k, v in fields.items()},
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 351, in <dictcomp>
    {k: self._generate_md_field_schema(k, v, decorators) for k, v in fields.items()},
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 614, in _generate_md_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 670, in _common_field_schema
    schema = apply_each_item_validators(schema, each_item_validators)
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 148, in apply_each_item_validators
    schema['schema'] = apply_each_item_validators(schema['schema'], each_item_validators)
  File "/tmp/docs/env/lib/python3.9/site-packages/pydantic/_internal/_generate_schema.py", line 163, in apply_each_item_validators
    raise TypeError(
TypeError: `@validator(..., each_item=True)` cannot be applied to fields with a schema of tuple-positional
```
It seems to step from importing the module here? I can give more detail / analysis of the above if you like - just tired atm! :laughing: 

### How to Reproduce

I'm installing these pip requirements.txt

```
sphinx
markupsafe
sphinx_immaterial
numpy
pydantic==1.10.8
pandas
matplotlib
nbsphinx
jupyter_client
notebook
nbconvert
ipykernel
recommonmark
sphinx_markdown_tables
sphinx-copybutton
readthedocs-sphinx-search
myst-parser
pymdown-extensions
```
At their latest versions as of tonight!

### Environment Information

```text
See error message above.
```

### Additional context

The fix for the error was installing `pydantic==1.10.8`

**Comment by picnixz** [comment]
Hi,

the issue is probably due to `sphinx_immaterial` as shown in the traceback:

>   File "/tmp/docs/env/lib/python3.9/site-packages/sphinx_immaterial/__init__.py", line 313, in setup
    app.setup_extension("sphinx_immaterial.custom_admonitions")

More precisely, it is the extension `custom_admonitions` from `sphinx_immaterial`. For that, I would advise opening an issue on their repository instead.

**Comment by vsoch** [comment]
Gotcha, thanks will do!

**Comment by 2bndy5** [comment]
For passersby, this is a duplicate of jbms/sphinx-immaterial#268 which was resolved and released in sphinx-immaterial v0.11.5.
